cmd/cli/kubectl-kyverno/commands/test/test.go

```diff
@@ -98,6 +98,51 @@ func runTest(out io.Writer, testCase test.TestCase, registryAccess bool, auditWa
 	} else {
 		fmt.Fprintln(out, "  Applying", len(policies)+len(validatingAdmissionPolicies), pluralize.Pluralize(len(policies)+len(validatingAdmissionPolicies), "policy", "policies"), "to", len(uniques), pluralize.Pluralize(len(uniques), "resource", "resources"), "...")
 	}
+	rulesmap := make(map[string]string)
+	policymap := make(map[string]string)
+	for _, res := range testCase.Test.Results {
+		if res.IsValidatingAdmissionPolicy {
+			continue
+		}
+		var ns string
+		var n bool
+		r := false
+		for _, policy := range policies {
+			if policy.IsNamespaced() {
+				ns = policy.GetNamespace()
+			}
+			if res.Policy == policy.GetName() || res.Policy == "default/"+policy.GetName() || res.Policy == ns+"/"+policy.GetName() {
+				n = true
+				for _, rule := range autogen.ComputeRules(policy) {
+					if res.Rule == rule.Name {
+						r = true
+						continue
+					}
+				}
+			}
+		}
+		if !r {
+			rulesmap[res.Policy] = res.Rule
+		}
+		if !n {
+			policymap[res.Policy] = res.Policy
+		}
+	}
+	var s string
+	if len(policymap) != 0 || len(rulesmap) != 0 {
+		if len(policymap) != 0 {
+			fmt.Printf("Error: ")
+			for _, policy := range policymap {
+				s = s + fmt.Sprintf("The policy named %v cannot be found.\n", policy)
+			}
```

Comment:
You could use `strings.Join()` to have a list that contains all unknown policies and then return the error.
It could be something similar to this:
```
if len(unknownPolicies) != 0 {
   result := strings.Join(unknownPolicies, ",")
   return nil, fmt.Errorf("Error: the following policies aren't found: %s", result)
}
```

Reply:
Ok I will update it

Reply:
We need to format the values so string.Join will be not good